### PR TITLE
legcord: new port

### PR DESCRIPTION
**Pull request by akierig**
#### Description

Legcord (originally Armcord) is a version of the Discord web client packaged for desktop. Unlike the Discord application or web client, it blocks Discord's trackers and consumes far fewer resources. It also supports all of the major Discord client plugin/theming frameworks.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 15.2 24C101 arm64
Xcode 16.2 16C5032a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by akierig** [comment]
advice would be appreciated here please.

tl;dr: it builds a `.app` in the location `dist/mac-${build_arch}` folder. on ARM machines, this is `mac-arm64` unfortunately on x86 (at least per the github builder logs) this comes out to `mac-x86_64` but the electron-builder folder is just `mac-x86.`

should I add in some 'if build_arch is x86_64 then move it from mac-x86' or similar?

**Comment by akierig** [comment]
I was privately reminded that legcord was originally designed and is optimized for arm64/aarch64 and have therefore restricted it to arm64 only.

File: net/Legcord/Portfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        Legcord Legcord 1.0.6 v
+github.tarball_from archive
+distname            ${github.project}-${github.version}
```
Comment (reneeotten) [inline net/Legcord/Portfile:8]: does the default not work?

Comment (akierig) [reply net/Legcord/Portfile:7]: it didn't the first time I tried it, but just trying it now, it does just fine. Must have been something I was doing. Yanked the 'archive' line

Comment (reneeotten) [reply net/Legcord/Portfile:7]: You need the “github,tarball_from” line; the “distname” should be redundabt I think

Comment (akierig) [reply net/Legcord/Portfile]: well, for some reason it works inconsistently on `...from releases`. so I switched back to archive and yanked the distname line.

sorry for the running around in circles.
